configuration_test.py

```diff
@@ -0,0 +1,189 @@
+        if not os.path.isfile(f):
+            raise PrerequisiteError("%s file does not exist. Please create it." % f)
+
+def check_cache(cache):
+    if not os.path.isdir(cache):
+        raise PrerequisiteError("Cache directory: (%s) \n\t\t   is not created. " % cache + \
+                                "Please manually create before running." )
+
+def check_stata(ARGUMENTS):
+    import yaml
+    import gslab_scons.misc as misc
+
+    sf_configs = load_yaml_value("user-config.yaml", "stata_executable")
+    sf         = ARGUMENTS.get('sf', sf_configs) 
+
+    # Fake scons-like env dict for misc.get_stata_executable(env)
+    fake_env = {'user_flavor': sf} 
+    stata_exec = misc.get_stata_executable(fake_env)
+    
+    if stata_exec is None:
+        raise PrerequisiteError('Stata is not installed or executable is not added to path')
+    
+    command = misc.get_stata_command(stata_exec)
+    check_stata_packages(command)
+    return sf
+
+def load_yaml_value(path, key):
+    import yaml
+
+    if key == "stata_executable":
+        prompt = "Enter %s value or None to search for defaults: "
+    else:
+        prompt = "Enter %s value: "
+
+    # Check if file exists and is not corrupted. 
+    # If so, load yaml contents.
+    yaml_contents = None
+    if os.path.isfile(path):
+        try:
+            yaml_contents = yaml.load(open(path, 'rU'))
+        except yaml.scanner.ScannerError:
+            message  = "%s is a corrupted yaml file. Delete file and recreate? (y/n) "
+            response = str(raw_input(message % path))
+            if response.lower() == 'y':
+                os.remove(path)
+                yaml_contents = None
+            else:
+                raise PrerequisiteError("%s is a corrupted yaml file. Please fix." % path)
+
+    # If key exists, return value.
+    # Otherwise, add key-value to file.
+    try:
+        return yaml_contents[key]
+    except:
+        with open(path, 'ab') as f:        
+            val = str(raw_input(prompt % key))
+            if val.lower() == "none":
+                val = None
+            else:
+                f.write('\n%s: %s\n' % (key, val))
+        return val
+
+def check_stata_packages(command):
+    import gslab_scons.misc as misc
+    if misc.is_unix():
+        command = command.split("%s")[0]
+    elif sys.platform == "win32":
+        command = command.split("do")[0]
+
+    try:
+        for pkg in ['yaml']:
+            call = (command + "which %s") % pkg
+            subprocess.check_output(call, stderr = subprocess.STDOUT, shell = True) # http://www.stata.com/statalist/archive/2009-12/msg00493.html and http://stackoverflow.com/questions/18962785/oserror-errno-2-no-such-file-or-directory-while-using-python-subprocess-in-dj
+            with open('stata.log', 'rU') as f:
+                log = f.read()
+    except subprocess.CalledProcessError:
+        raise PrerequisiteError("Please confirm stata command (%s) is correct." % command)
```

Comment:
I don't like this wording... if this message appears, the Stata command is probably incorrect.